### ISSUE TITLE
 Don't tag on every push

### ISSUE THREAD

**Issue by pasbi** [pasbi/ommpfritt#121]
In order to create a release after every push, I somehow told github (or maybe Travis-CI) to automatically create a tag on the last pushed commit.

That's actually pretty annoying, so I'd like to disable this again, but:
- I still want to provide build artefacts for the latest build
- I don't find the button to turn off the auto-tag anymore.

I think @probonopd knows what to do. Iirc, AppImages we're provided for download even for untagged commits.

**Comment by probonopd** [comment]
https://github.com/pasbi/ommpfritt/blob/master/build-scripts/upload-appimage.sh is uploading each continuous build to GitHub Releases. For GitHub Releases to match the SHA of the commit, `uploadtool` deletes the `continuous` tag and creates a new one, to which the build artifacts will be uploaded.

If you do not like this, you could let https://github.com/tcnksm/ghr/ do the uploading using `-replace # Replace artifacts if it is already uploaded`.  This way, the existing tag will be re-used and no new one is created. This, however, has the downside that the displayed SHA and date displayed on GitHub Releases will not match the actual contents of the AppImage.

**Comment by pasbi** [comment]
I like that there was a `continuous` tag on the tip of the tree and I'd like to restore that behaviour.

Currently, a tag appears after every push, which is named like `untagged-HASH`. The problem is, that it persists.
In the meanwhile, the `continuous` tag has not been touched since 12 days, which I find suspicious.

https://github.com/pasbi/ommpfritt/tags

I haven't really investigated that issue (low prio and I didn't know where to start).
I will have a look into what the `uploadtool` is doing.

**Comment by pasbi** [comment]
The tags are created because of the deployment section in `.travis.yml`-file.
The problem is now:
1. The continuous tag is not updated,
2. I cannot deploy the windows installer.

**Comment by probonopd** [comment]
> I like that there was a continuous tag on the tip of the tree and I'd like to restore that behaviour.

`ghr` can do this as well:

```
-delete \         # Delete release and its git tag in advance if it exists (same as -recreate)
```

**Comment by pasbi** [comment]
Update:
A new tag is now created on every commit (`continuous-$HASH`).
I'm afraid that deleting (or moving) a tag may cause troubles, for example with VSCode or other editors that have git built-in.

[I guess it's common sense that re-writing history in git will screw up workflow with collaborators](https://stackoverflow.com/a/25849917), but maybe you have arguments why this practice is safe in this special case?

> ghr can do this as well:

I guess you refer to GitHub Releases (sorry, I'm quite unfamiliar with the abbreviations)?
Where should I add that line? In `.travis.yml`'s `deploy`-section?

**Comment by probonopd** [comment]
>> ghr can do this as well:

> I guess you refer to GitHub Releases (sorry, I'm quite unfamiliar with the abbreviations)?

https://github.com/tcnksm/ghr/

**Comment by pasbi** [comment]
I redesigned the whole CI pipeline and this is no longer relevant.

FTR: releases must be done manually now (use [make-release.sh](https://github.com/pasbi/ommpfritt/blob/main/build-scripts/make_release.sh)).
Continuous releases are now available as workflow artifacts, if someone needs such special releases I expect them to be able to find them in the actions section.

@probonopd I managed to build the AppImage, but it does not yet get uploaded to the AppImageHub.
I plan to implement that (later...), although I don't know how to do that at the moment. I'd appreciate a PR :smiley:
